src/qml/QFieldGuide.qml

```diff
@@ -0,0 +1,327 @@
+import QtQuick
+import QtQuick.Controls
+import QtQuick.Shapes
+import QtQuick.Window
+import QtQuick.Controls.Material
+import Theme
+
+Popup {
+  id: guide
+  padding: 0
+  parent: Overlay.overlay
+  width: internalObject.parentWidth
+  height: internalObject.parentHeight
+  background: Item {
+  }
+  contentItem: Item {
+  }
+  clip: true
+  onVisibleChanged: {
+    if (visible) {
+      guide.index = 0;
+      enablePannelAnimation = false;
+    }
+  }
+  onIndexChanged: {
+    canvas.requestPaint();
+    if (index == 1)
+      enablePannelAnimation = true;
+  }
+
+  property bool enablePannelAnimation: false
+  property var steps: []
+  property int targetMargins: 5
+  property Component nextButton: nextButton
+  property Component prevButton: prevButton
+  property int index: 0
+  property string finishText: qsTr("Finish")
+  property string nextText: qsTr("Next")
+  property string previousText: qsTr("Previous")
+
+  Component {
+    id: nextButton
+    QfButton {
+      text: isEnd ? guide.finishText : guide.nextText
```

Comment:
Could we rename this to isLast / isLastItem / isLastText?

Reply:
done :+1: